src/clp_logging/handlers.py

```diff
@@ -408,10 +417,13 @@ def log_fn(msg: str) -> None:
                 last_timestamp_ms = CLPEncoder.encode_timestamp(last_timestamp_ms, buf)
                 if loglevel_timeout:
                     loglevel_timeout.update(loglevel, last_timestamp_ms, log_fn)
-                zstream.write(buf)
+                ostream.write(buf)
             if loglevel_timeout:
                 loglevel_timeout.timeout()
-            zstream.write(EOF_CHAR)
+            ostream.write(EOF_CHAR)
+
+            if zstd_compress:
```

Comment:
Please add a comment on why it is difficult to use context manager and the reason why we need to add a close here explicitly.